src/vmm/src/vstate/memory.rs

```diff
@@ -485,6 +493,25 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_from_state() {
+        let state = GuestMemoryState {
+            regions: vec![GuestMemoryRegionState {
+                base_address: 0,
+                size: 4096,
+                offset: 0,
+            }],
+        };
+        let file = TempFile::new().unwrap().into_file();
+
+        // No mapping of snapshots that were taken with hugetlbfs enabled
+        let err =
+            GuestMemoryMmap::from_state(Some(&file), &state, false, HugePageConfig::Hugetlbfs2M)
+                .unwrap_err();
+
+        assert!(matches!(err, MemoryError::HugetlbfsSnapshot), "{:?}", err);
```

Comment:
It does need `matches` because `assert_eq` requires `MemoryError` to implement `PartialEq`, which it cannot because it contains an `std::io::Error` :(